sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/core/ShareTest.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.share.perf.core;
+
+import com.azure.perf.test.core.PerfStressOptions;
+import com.azure.storage.file.share.ShareAsyncClient;
+import com.azure.storage.file.share.ShareClient;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+public abstract class ShareTest<TOptions extends PerfStressOptions> extends ServiceTest<TOptions> {
+    private static final String SHARE_NAME = "perfstress-sharev11-" + UUID.randomUUID().toString();
+
+    protected final ShareClient shareClient;
+    protected final ShareAsyncClient shareAsyncClient;
+
+    public ShareTest(TOptions options) {
+        super(options);
+        // Setup the container clients
+        String shareName = "perfstress-sharev11-" + UUID.randomUUID().toString();
+        shareClient = shareServiceClient.getShareClient(shareName);
+        shareAsyncClient = shareServiceAsyncClient.getShareAsyncClient(shareName);
+    }
+
+    @Override
+    public Mono<Void> setupAsync() {
+        return super.setupAsync().then(Mono.defer(() -> {
```

Comment:
Can this use the same (and simpler) pattern as DataLake?

https://github.com/Azure/azure-sdk-for-java/pull/17883/files#diff-10b82805ef0ae91ec67e7271940bf4767256bb9e37a9833e7f27d343332eb40aR29